.github/workflows/publish_docker.yml

```diff
@@ -0,0 +1,41 @@
+name: publish docker
+
+on:
+  push:
+    paths:
+      - "src/**"
+      - ".github/workflows/publish_docker.yml"
+    branches:
+      - main
+  pull_request:
+    paths:
+      - "src/**"
+      - ".github/workflows/publish_docker.yml"
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        include:
+          - tag: "7.0" #Debian 11
+            file: Dockerfile
+          - tag: 7.0-jammy # Ubuntu 22.04
+            file: Dockerfile
+          - tag: 7.0-alpine # Alpine 3.17
+            file: Dockerfile.alpine
+          - tag: 7.0-alpine3.16
+            file: Dockerfile.alpine
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - name: checkout
+        uses: Brightspace/third-party-actions@actions/checkout
+      - name: setup docker buildx
+        uses: docker/setup-buildx-action@v2
```

Comment:
Do we need docker buildx here? I thought that's only for cross-arch builds.
Cross-arch builds would be nice for sure, I just want to confirm if we are using some other features of buildx, because we aren't building across arch here.

Reply:
buildx is for the `-o` flag.  On my local machine I could have just `build` work plus setting the $DOCKER_BUILDKIT env var set to 1 but it wasn't saving to the runner's disk.  Thought it would be easier to have the binary on the runner and not do anything with volumes. I think buildx is just to use the buildkit backend instead of the default

Reply:
Yea I guess what I did is the equivalent as just doing `buildx ...`. This flag only works when using the buildkit backend https://docs.docker.com/engine/reference/commandline/build/#output